plugins/inputs/sqlserver/azuresqlqueries.go

```diff
@@ -389,6 +389,130 @@ HAVING
 OPTION(RECOMPILE);
 `
 
+const sqlAzureDBPartQueryPeriod = `
+DECLARE @QueryData NVARCHAR(MAX) = ?1; --input parameter
+
+IF SERVERPROPERTY('EngineEdition') <> 5 BEGIN /*not Azure SQL DB*/
+	DECLARE @ErrorMessage AS nvarchar(500) = 'Telegraf - Connection string Server:'+ @@SERVERNAME + ',Database:' + DB_NAME() +' is not an Azure SQL DB. Check the database_type parameter in the telegraf configuration.';
+	RAISERROR (@ErrorMessage,11,1)
+	RETURN
+END;
+
+DECLARE @lastIntervalEndTimestamp BIGINT = 0;
+
+/* Get the time when the query was previously called */
+IF @QueryData != '' AND @QueryData IS NOT NULL
+	SET @lastIntervalEndTimestamp = CAST(@QueryData AS BIGINT);
+
+DECLARE @secondsInDay INT = 86400;
+
+/* Get query aggregation interval from query store */
+DECLARE @timeGrain BIGINT = (SELECT interval_length_minutes * 60 FROM sys.database_query_store_options WITH (NOLOCK)); 
+
+/* Get current time and round it down to a multiple of @timeGrain */
+DECLARE @currDate DATETIME = GETUTCDATE();
+DECLARE @currIntervalEndTimestamp BIGINT = DATEDIFF_BIG(second, 0, @currDate) / @timeGrain * @timeGrain;
+
+/* If we are still in the same @timeGrain interval with when the query was previously called, quit and do not report any data */
+IF @lastIntervalEndTimestamp = @currIntervalEndTimestamp
+	RETURN;
+
+/* Calculate the start time for returning data from query store:
+If the query was never called before, start time is end time minus @timeGrain */
+DECLARE @currIntervalStartTimestamp BIGINT = @currIntervalEndTimestamp - @timeGrain;
+
+/* If the query was called before, start time is when the query was previously called, so that we return all the records since the previous call */
+IF @lastIntervalEndTimestamp != 0
+	SET @currIntervalStartTimestamp = @lastIntervalEndTimestamp;
+
+/* Convert start and end times to datetime */
+DECLARE @currIntervalEndDate datetime = DATEADD(second, @currIntervalEndTimestamp % @secondsInDay, DATEADD(day, @currIntervalEndTimestamp / @secondsInDay, 0));
+DECLARE @currIntervalStartDate datetime = DATEADD(second, @currIntervalStartTimestamp % @secondsInDay, DATEADD(day, @currIntervalStartTimestamp / @secondsInDay, 0));
+
```

Comment:
@dimitri-furman Is the recommendation to use datetime2 at all places? datetime2 seems to have higher precision than datetime, however it is not aware of timezone offset.

Reply:
Query Store uses datetimeoffset in sys.query_store_runtime_stats_interval. We should stay consistent and use the same data type to avoid type conversion problems. But you should check if the current time math that uses datetime still works correctly with datetimeoffset.